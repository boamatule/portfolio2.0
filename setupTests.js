import Enzyme from 'enzyme';
import Adapter from '@cfaester/enzyme-adapter-react-18';

Enzyme.configure({ adapter: new Adapter() });


// import { configure } from 'enzyme';
// import Adapter from '@cfaester/enzyme-adapter-react-18';

// configure({ adapter: new Adapter() });