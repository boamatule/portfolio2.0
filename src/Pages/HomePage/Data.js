/* eslint-disable import/prefer-default-export */
export const homeObjOne = {
  lightBg: true,
  primary: false,
  imgStart: '',
  lightTopLine: true,
  lightTextDesc: true,
  buttonLabel: 'Check my work',
  description: 'Hello, Boa was born and raised in Mozambique. Currently based in Oslo after almost 7 years in Sweden.',
  headline: 'Full Stack Developer',
  lightText: false,
  // topLine: 'Love Yourself',
  img: '/images/avatar.svg',
  alt: 'avatar',
  start: '',
};

export const homeObjTwo = {
  lightBg: true,
  primary: false,
  imgStart: 'start',
  lightTopLine: true,
  lightTextDesc: true,
  buttonLabel: 'Visit the page!',
  description: 'Boa was born and raised in Mozambique. Currently based in Oslo.',
  headline: 'The Legends of the century',
  lightText: false,
  topLine: 'Love Yourself',
  img: '/images/avatar.svg',
  alt: 'Image',
  start: 'true',
};

export const homeObjThree = {
  lightBg: false,
  primary: true,
  imgStart: 'start',
  lightTopLine: true,
  lightTextDesc: true,
  buttonLabel: 'Join us now!',
  description: 'Boa was born and raised in Mozambique. Currently based in Oslo after almost 7 years in Sweden',
  headline: 'The Legends of the century',
  lightText: false,
  topLine: 'Love Yourself',
  img: '/images/avatar.svg',
  alt: 'Image',
  start: 'true',
};

export const homeObjFour = {
  lightBg: false,
  primary: true,
  imgStart: 'start',
  lightTopLine: true,
  lightTextDesc: true,
  buttonLabel: 'Join us now!',
  description: 'Boa was born and raised in Mozambique. Currently based in Oslo after almost 7 years in Sweden',
  headline: 'The Legends of the century',
  lightText: false,
  topLine: 'Love Yourself',
  img: '/images/cooper.svg',
  alt: 'Image',
  start: '',
};
