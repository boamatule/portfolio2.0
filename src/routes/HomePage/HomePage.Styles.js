import styled from "styled-components/macro";
import { Container } from "../../global-styles";

export const HomeContainer = styled(Container)`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-between;
  
  @media screen and (max-width: 960px) {
    margin: 0 auto;
    padding: 0 4px;
  }
`;

export const HomeWrapper = styled.div`
  margin-top: 80px;
`;

export const ProjectsListContainer = styled.div`
  margin-top: auto;
  margin-bottom: 60px;
  justify-content: space-between;
`;

