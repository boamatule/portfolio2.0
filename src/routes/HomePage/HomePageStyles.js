import styled from 'styled-components/macro';
import { Link } from 'react-router-dom';
import { motion } from 'framer-motion';
import { Container } from '../../global-styles';

export const  HomePageWrapper = styled(Container)`
  /* display: flex;
  justify-content: space-around;
  align-items: center; */
  margin-top: 120px;
  /* position: absolute; */

  @media (max-width: 960px) {
    margin-right: auto;
    margin-left: auto;
    padding-right: 4px;
    padding-left: 4px;
  }
`;