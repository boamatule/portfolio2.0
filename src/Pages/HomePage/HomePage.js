import React, { useState } from 'react';
import { HomeInfo, Profile } from '../../components';
import ContactPage from '../ContactPage/ContactPage';
import PortfolioListPage from '../PortfolioPage/PortfolioListPage';
import { homeObjOne, homeObjTwo, homeObjThree, homeObjFour } from './Data';
import { Container } from '../../global-styles';

const HomePage = () => (
  <Container>
    <Profile />
    <PortfolioListPage />
    {/* <ContactPage /> */}
  </Container>
);

export default HomePage;
