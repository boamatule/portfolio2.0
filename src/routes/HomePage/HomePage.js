import React from 'react';
import About from '../../Pages/AboutPage/AboutPage';
import ContactPage from '../../Pages/ContactPage/ContactPage';
import ProjectsList from '../../components/ProjectsList/ProjectsList';
import { HomeContainer } from './HomePage.Styles';
import Profile from '../../components/Profile/Profile';
import styled from 'styled-components';
import tw from 'twin.macro';

const HomePageContaner  = styled.div`
${tw`flex flex-col items-center w-full h-full overflow-x-hidden`}
`;

const HomePage = () => {
  return (
    <HomePageContaner>
      {/* <Profile /> */}
      <About />
      <ProjectsList />
      <ContactPage />
    </HomePageContaner>
  );
}
export default React.memo(HomePage);
