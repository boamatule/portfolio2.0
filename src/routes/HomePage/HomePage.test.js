import { shallow } from 'enzyme';
import React from 'react';
import '../../../setupTests';
import HomePage from './HomePage';

it('expect to render NavBar Component', () => {
  expect(shallow(<HomePage />)).toMatchSnapshot();
});

