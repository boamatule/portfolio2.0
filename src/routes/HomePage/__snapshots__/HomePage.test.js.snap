// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render NavBar Component 1`] = `ShallowWrapper {}`;
